applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -357,6 +357,21 @@ namespace Kratos {
         KRATOS_CATCH("")
     }
 
+    void ContinuumExplicitSolverStrategy::ComputeNewRigidFaceNeighboursHistoricalData() {
+        KRATOS_TRY
+
+        BaseType::ComputeNewRigidFaceNeighboursHistoricalData();
+
+        const int number_of_particles = (int) mListOfSphericContinuumParticles.size();
+
+        #pragma omp parallel for
+        for (int i = 0; i < number_of_particles; i++) {
+            mListOfSphericContinuumParticles[i]->ReorderFEMneighbours();
+        }
```

Comment:
parallel_utils :)

Reply:
`IndexPartition<unsigned int>(mListOfSphericContinuumParticles.size()).for_each([&](unsigned int i){
            mListOfSphericContinuumParticles[i]->ReorderFEMneighbours();
        });`
maybe

Reply:
why not `block_for_each` ? What's the difference between those two?

Reply:
According to [parallelUtils](https://github.com/KratosMultiphysics/Kratos/wiki/Parallel-Utilities) it seems more adequate to use index based loop.  Not sure if it can be also done with simple block_for_each loop. @philbucher 

```
IndexPartition<unsigned int>(data_vector.size()).for_each([&](unsigned int i){
    output[i] = std::pow(data_vector[i], 0.01);
});

block_for_each(model_part.Nodes(), [&](Node<3>& rNode){
    noalias(rNode.FastGetSolutionStepValue(rVariable)) = value;
});
```